packages/jsii-diff/lib/util.ts

```diff
@@ -0,0 +1,69 @@
+import child_process = require('child_process');
+import fs = require('fs-extra');
+import log4js = require('log4js');
+import os = require('os');
+import path = require('path');
+import tar = require('tar-fs');
+import util = require('util');
+
+const LOG = log4js.getLogger('jsii-diff');
+
+const exec = util.promisify(child_process.exec);
+
+export async function inTempDir<T>(block: () => Promise<T>): Promise<T> {
+  const origDir = process.cwd();
```

Comment:
I like to call such methods “withXxX”

Reply:
Agreed, but in this case I feel that doesn't cover the semantics completely, as the block does a `chdir` instead of just providing a value.

So the block doesn't happen "*with*" a value, instead it happens "*in*" a directory.